server/app/assets/javascripts/admin_question_edit.ts

```diff
@@ -232,6 +233,28 @@ class AdminQuestionEdit {
       addButton.classList.remove('hidden')
     })
   }
+
+  /**
+   * Clears validation errors when a user selects a new key for map question settings.
+   * Removes error message and error classes from wrapper when a key is changed.
+   */
+  addMapKeyErrorHandler() {
+    document.addEventListener('change', (event: Event) => {
```

Comment:
suggestion: could you add an event listener just to the `data-key-select` element if it exists so this method doesn't fire on every change?